src/table/use-column-widths.tsx

```diff
@@ -178,9 +178,10 @@ export function ColumnWidthsProvider({ visibleColumns, resizableColumns, contain
       return;
     }
     setColumnWidths(() => readWidths(getCell, visibleColumns));
-    // This code is intended to run only at the first render and should not re-run when table props change
+    // This code is intended to run only at the first render (or when rendering a column that was not previously rendered)
+    // and should otherwise not re-run when table props change
```

Comment:
The `visibleColumns` is not memoized so now this effect that was previously used in the initial render only is executed for every render, e.g. when a column was resized or a row was selected.

Reply:
Simplified this logic to combine the two `useEffect` hooks that were doing similar but subtly different things.